lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -300,6 +300,11 @@ void CircuitLoweringState::processRemainingAnnotations(
     // This can occur for example if an annotation marks something in the IR as
     // not to be processed by a pass, but that pass hasn't run anyway.
     if (a.isClass(
+            // If the class is `circt.nonlocal`, it's not really an annotation,
+            // but part of a path specifier for another annotation which is
+            // non-local.  We can ignore these path specifiers since there will
+            // be a warning produced for the real annotation.
+            "circt.nonlocal",
```

Comment:
I think this is fine for now.  We probably want a function at some point that removes an NLA and all the back pointers annotations.